### PR TITLE
Make artifact built-by buildrecord relationship N-1

### PR DESCRIPTION
**Pull request by janinko**
### Checklist:

* [x] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

**Comment by janinko** [comment]
retest this please

**Comment by janinko** [comment]
> I missed the failing tests. You should remove
> 
> https://github.com/project-ncl/pnc/blob/8833751ce45ebab74a1a6b3229a841262ad19e25/model/src/test/java/org/jboss/pnc/model/CacheHandlerTest.java#L190
> 
> as the cache does not exist anymore

Removed

**Comment by janinko** [comment]
retest this please

File: bpm/src/test/java/org/jboss/pnc/bpm/test/BuildResultPushManagerTest.java [hunk 1]
```diff
@@ -170,7 +170,6 @@ private BuildRecord buildRecord() {
         record.setId(buildRecordIdSequence++);
         record.setStatus(BuildStatus.SUCCESS);
         record.setBuildConfigurationAudited(bca);
-        record.setBuiltArtifacts(Collections.emptySet());
```
Comment (jbartece) [inline bpm/src/test/java/org/jboss/pnc/bpm/test/BuildResultPushManagerTest.java]: Why did you remove this line?
